hw16/Player.cpp

```diff
@@ -0,0 +1,51 @@
+#include "Player.h"
+#include "Weapon.h"
+#include "Team.h"
+#include <iostream>
+
+Player::Player(const std::string& name, int health, PlayerClass playerClass)
+        : m_name(name), m_health(health), m_playerClass(playerClass), m_weapon(nullptr), m_team(nullptr) {}
+
+void Player::setWeapon(Weapon* weapon) {
+    m_weapon = weapon;
+}
+
+void Player::setTeam(Team* team) {
+    m_team = team;
+}
+
+Weapon* Player::getWeapon() const {
+    return m_weapon;
+}
+
+std::string Player::getPlayerClassName() const {
+    switch (m_playerClass) {
+        case ASSAULT:
+            return "Assault";
+        case SUPPORT:
+            return "Support";
+        case SNIPER:
+            return "Sniper";
+        case MEDIC:
+            return "Medic";
+        default:
+            return "Unknown";
+    }
+}
+
+void Player::printPlayerInfo() const {
+    std::cout << "| Player q._.p |" << std::endl
+              << "Name: " << m_name << std::endl
+              << "Health: " << m_health << std::endl
+              << "Class: " << getPlayerClassName() << std::endl;
+
+    if (m_weapon != nullptr) {
+        m_weapon->getInfo();
+    }
+
+    if (m_team != nullptr) {
+        m_team->getInfo(false);
```

Comment:
See, here without looking at implementation of Team::getInfo(bool) its not clear what this true/false value stands for
having two distinct method and calling here m_team->getTeamName() would sort this out